src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -1200,6 +1200,17 @@ private static SyntaxNode WithAttributeLists(SyntaxNode declaration, SyntaxList<
             _ => declaration,
         };
 
+    internal override ImmutableArray<SyntaxNode> GetPrimaryCtor(SyntaxNode declaration)
+    {
+        if (declaration is ClassDeclarationSyntax || declaration is StructDeclarationSyntax)
```

Comment:
can just check for `TypeDeclarationSyntax`

Reply:
Done